hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -0,0 +1,249 @@
+    private final List<HederaAccount> accountsCreated = new ArrayList<>();
+    private AccountNumbers accountNumbers;
+
+    /**
+     * @param evmAddress the EVM address of the blocked account
+     * @param memo      the memo of the blocked account
+     */
+    record BlockedInfo(ByteString evmAddress, String memo) {}
+
+    @Inject
+    public BlocklistAccountCreator(
+            final Supplier<HederaAccount> accountSupplier,
+            final EntityIdSource ids,
+            final BackingStore<AccountID, HederaAccount> accounts,
+            final Supplier<JEd25519Key> genesisKeySource,
+            final @CompositeProps PropertySource properties,
+            final AliasManager aliasManager,
+            AccountNumbers accountNumbers) {
+        this.blocklistFileName = properties.getStringProperty(ACCOUNTS_BLOCKLIST_FILE);
+        this.accountSupplier = accountSupplier;
+        this.ids = ids;
+        this.accounts = accounts;
+        this.genesisKeySource = genesisKeySource;
+        this.properties = properties;
+        this.aliasManager = aliasManager;
+        this.accountNumbers = accountNumbers;
+    }
+
+    /**
+     * Makes sure that all blocked accounts contained in the blocklist file are present in state, and creates them if necessary.
+     */
+    public void createMissingAccounts() {
+        final List<BlockedInfo> blocklist;
+        try {
+            final var fileLines = readPrivateKeyBlocklist(blocklistFileName);
+            final var columnCount = fileLines.get(0).split(",").length;
+            blocklist = fileLines.stream()
+                    .skip(1)
+                    .map(line -> parseCSVLine(line, columnCount))
+                    .toList();
+        } catch (IllegalArgumentException iae) {
+            log.error("Failed to parse blocklist, entry does not have required number of columns", iae);
+            return;
+        } catch (DecoderException de) {
+            log.error("Failed to parse blocklist, entry not in hex format", de);
+            return;
+        } catch (Exception e) {
+            log.error("Failed to read blocklist file {}", blocklistFileName, e);
+            return;
+        }
+
+        final var blockedToCreate = blocklist.stream()
+                .filter(blockedAccount ->
+                        aliasManager.lookupIdBy(blockedAccount.evmAddress).equals(MISSING_NUM))
+                .collect(Collectors.toSet());
+
+        final var genesisAccountId = AccountID.newBuilder()
+                .setRealmNum(0)
+                .setShardNum(0)
+                .setAccountNum(accountNumbers.treasury())
+                .build();
+
+        for (final var blockedInfo : blockedToCreate) {
+            final var newId = ids.newAccountId(genesisAccountId); // get the next available new account ID
+            final var account = blockedAccountWith(blockedInfo);
+            accounts.put(newId, account); // add the account with the corresponding newId to state
```

Comment:
In theory it can fail if we pass null, but I don't see that happening here and also I have not seen exceptional cases handled anywhere in the codebase for `BackingAccounts.put()`.